### DISCUSSION TITLE
Best practices for using recipes

### DISCUSSION THREAD

**Discussion by Ca-ressemble-a-du-fake** [coqui-ai/TTS#2033] [General Q&A] [answered]
Hi, 

I am used to using config.json file to configure the training of each speaker I want to train on. This is handy for me because I can store the config.json files along side their matching dataset.
But now I want to train multi-lingual / multi speakers YourTTS model and need to use the recipe.

To make it work I run the recipe from the coqui/TTS root folder but I found it "dangerous" to mix code with this recipe modified to meet my needs (that I want to keep to use for other projects). I wrote dangerous because I could update the TTS folder from github and loose those modified recipes.

That's why I am asking if there are some best practices when using recipes regarding where to store them and etc. 

On solution could be to modify the [Python path](https://stackoverflow.com/a/4383597/6351897) but it may have undesired effects on training / inference that I would prefer to avoid :D!

Thanks in advance for your suggestions,

**Comment by AnykeyXD** [comment]
create your own version by either creating a new branch or copying the file into your own recipe folder. look up version control and its purpose.

**Reply by Ca-ressemble-a-du-fake** [reply]
Thank you @AnykeyXD . If I create my own separate recipe folder then I cannot run them from this folder because of the different imports at the beginning of the recipes, can I ? That's why I am running the recipe from coqui/TTS/TTS folder. But I may have missed something!

**Reply by lexkoro** [reply]
If you install TTS `pip install -e .` ([training models](https://github.com/coqui-ai/TTS#install-tts)) it shouldn't matter from where you run the recipe.

**Reply by Ca-ressemble-a-du-fake** [reply]
@lexkoro interesting thank you! Then I may have screwed up something. I will reboot and try again!

**Reply by Ca-ressemble-a-du-fake** [reply]
That's correct @lexkoro, what a relief! I should have screw up my python configuration on my notebook because it actually works on the training machine (which was busy that's why I checked a new recipe on the notebook)!

**Comment by Ca-ressemble-a-du-fake** [comment, marked as answer]
To make it easier to find the solution without reading through all comments I write it again. When correctly installed it does not matter where the recipe is run from :

Quoting @lexkoro's [answer above](https://github.com/coqui-ai/TTS/discussions/2033#discussioncomment-3756359) : 

>  If you install TTS pip install -e . ([training models](https://github.com/coqui-ai/TTS#install-tts)) it shouldn't matter from where you run the recipe.
